From 61c43fc114e860c70e17a18e1524dc68cb7d0dc7 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Fri, 20 Sep 2019 19:39:57 +0300
Subject: [PATCH 214/245] enetc: Fix 10G PCS PHY access

commit 62788ebe6a35ec499aac50e4e6981c6bb7abdd31 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

Dev addr was missing in C45 access to PCS, without it the code writes to
dev 0 which is not implemented.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index 761e2c88aa83..f1e03a79d5e3 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -832,11 +832,12 @@ static void enetc_configure_sgmii(struct mii_bus *imdio)
 static void enetc_configure_sxgmii(struct mii_bus *imdio)
 {
 	/* Dev ability - SXGMII */
-	imdio->write(imdio, 0, ENETC_PCS_DEV_ABILITY | MII_ADDR_C45,
-		     ENETC_PCS_DEV_ABILITY_SXGMII);
+	imdio->write(imdio, 0, MII_ADDR_C45 | (MDIO_MMD_VEND2 << 16) |
+		     ENETC_PCS_DEV_ABILITY, ENETC_PCS_DEV_ABILITY_SXGMII);
 
 	/* Restart PCS AN */
-	imdio->write(imdio, 0, ENETC_PCS_CR | MII_ADDR_C45,
+	imdio->write(imdio, 0, MII_ADDR_C45 | (MDIO_MMD_VEND2 << 16) |
+		     ENETC_PCS_CR,
 		     ENETC_PCS_CR_LANE_RESET | ENETC_PCS_CR_RESET_AN);
 }
 
-- 
2.17.1

